modules/react-components/src/tree-view/tree-view.tsx

```diff
@@ -0,0 +1,350 @@
+            });
+        } else {
+            currentNode.isChecked = e.target.checked;
+            toggledNodes.push(currentNode);
+        }
+
+        onCheckToggleCb(toggledNodes, depth);
+        setLastCheckToggledNodeIndex(currentNodeIndex);
+        handleUpdate(data);
+    }
+
+    const handleDelete = (node: any) => {
+        const { onDeleteCb, depth } = props;
+        const data = _.cloneDeep(treeData);
+
+        const newData = data.filter((nodeItem) => {
+            return !_.isEqual(node, nodeItem);
+        });
+
+        onDeleteCb(node, newData, depth) && handleUpdate(newData);
+    }
+
+    const handleExpandToggle = (node: TreeNode) => {
+        const { onExpandToggleCb, depth } = props;
+        const data = _.cloneDeep(treeData);
+        const currentNode = _.find(data, node);
+
+        currentNode.isExpanded = !currentNode.isExpanded;
+
+        if (onExpandToggleCb) {
+            onExpandToggleCb(currentNode, depth);
+        }
+
+        handleUpdate(data);
+    }
+
+    const printCheckbox = (node: TreeNode) => {
+        const { isCheckable, keywordLabel, depth } = props;
+        const nodeText = _.get(node, keywordLabel, '');
+
+        if (isCheckable(node, depth)) {
+            return (
+                <label htmlFor={node.name} className="tree-label">
+                    <input
+                        type="checkbox"
+                        name={node[keywordLabel]}
+                        className="invisible"
+                        onClick={(e) => {
+                            handleCheckToggle(node, e);
+                        }}
+                        checked={!!node.isChecked}
+                        id={node.name}
+                    />
+                    <div className="checkbox">
+                        <svg width="17px" height="17px" viewBox="0 0 20 20">
+                            <path d="M3,1 L17,1 L17,1 C18.1045695,1 19,1.8954305 19,3 L19,17 L19,17 C19,
+                                18.1045695 18.1045695,19 17,19 L3,19 L3,19 C1.8954305,19 1,18.1045695 1,17 L1,3 L1,
+                                3 C1,1.8954305 1.8954305,1 3,1 Z"></path>
+                            <polyline points="4 11 8 15 16 6"></polyline>
+                        </svg>
+                    </div>
+                    <span>{nodeText}</span>
+                </label>
+            );
+        }
+    }
+
+    const printDeleteButton = (node: TreeNode) => {
+        const { isDeletable, depth, deleteElement } = props;
+
+        if (isDeletable(node, depth)) {
+            return (
+                <div className="delete-btn"
+                    onClick={() => {
+                        handleDelete(node);
+                    }}
+                >
+                    {deleteElement}
+                </div>
+            );
+        }
+    }
+
+    const printExpandButton = (node: TreeNode): ReactElement => {
+        const className = node.isExpanded
+            ? ''
```

Comment:
Any reason have this check in multiple lines?

Reply:
This was actually a code migrate from an existing plugin to make it easier to style as the way we want. Any how I have fixed this with commit 8afade7